SocialCareCaseViewerApi/V1/Gateways/ProcessDataGateway.cs

```diff
@@ -31,24 +31,39 @@ public ProcessDataGateway(ISccvDbContext sccvDbContext, ISccvDbContextTemp sccvD
         }
         public IEnumerable<CareCaseData> GetProcessData(ListCasesRequest request)
         {
-            var firstNameFilter = !string.IsNullOrWhiteSpace(request.FirstName) ?
-                Builders<BsonDocument>.Filter.Regex("first_name", BsonRegularExpression.Create(new Regex(request.FirstName, RegexOptions.IgnoreCase))) : null;
-            var lastNameFilter = !string.IsNullOrWhiteSpace(request.LastName) ?
-                Builders<BsonDocument>.Filter.Regex("last_name", BsonRegularExpression.Create(new Regex(request.LastName, RegexOptions.IgnoreCase))) : null;
-            var workerEmailFilter = !string.IsNullOrWhiteSpace(request.WorkerEmail) ?
-                Builders<BsonDocument>.Filter.Regex("worker_email", BsonRegularExpression.Create(new Regex(request.WorkerEmail, RegexOptions.IgnoreCase))) : null;
-            var caseNoteTypeFilter = !string.IsNullOrWhiteSpace(request.CaseNoteType) ?
-                Builders<BsonDocument>.Filter.Regex("case_note_type", BsonRegularExpression.Create(new Regex(request.CaseNoteType, RegexOptions.IgnoreCase))) : null;
+            List<BsonDocument> result;
+            long mosaicId = 0;
+            if (!string.IsNullOrEmpty(request.MosaicId))
+            {
+                if (!Int64.TryParse(request.MosaicId, out mosaicId))
+                {
+                    throw new Exception();
+                }
+                var filter = "{$or: [{ mosaic_id: " + mosaicId.ToString() + "}, { mosaic_id: /" + mosaicId.ToString() + "/}]}";
```

Comment:
is that 'or' required in the filter if we are setting it against single property?